### DISCUSSION TITLE
vscode fortls error

### DISCUSSION THREAD

**Discussion by Lucas-Morais-Freire** [krvajal/vscode-fortran-support#970] [Q&A - deprecated see Fortran Discourse] [answered]
When I install the "Modern Fortran" extension it asks me to install fortls and after I click "install" this popup appears:

Modern Fortran Error starting fortls: Check that fortls is in your PATH or that "fortran.fortls.path" is pointing to a fortls binary.

I have checked and the path is:

C:\Users\lucas\AppData\Roaming\Python\Python311\site-packages\fortls

which seems fine to me. Any possible reasons why its not working?

**Comment by gnikit** [comment]
For Windows, the autodetection for `fortls` in Modern Fortran can be spotty, I am working on fixing that.
Regardless, when specifying a path to a binary in Windows it is almost always accompanied from the `.exe` extension.
`pip` on Windows installs its binaries under a folder called `Scripts`, try locating that folder.

An educated guess on how the setting should look like:

```json
"fortran.fortls.path": "C:\\Users\\lucas\\AppData\\Roaming\\Python\\Python311\\Scripts\\fortls.exe"
```

**Reply by rushingfox** [reply]
It is really useful! For MACos user it also holds: but we need to specify the executable without the ".exe" extension.

This works for me:
"fortran.fortls.path": "/Users/myusername/Library/Python/3.9/bin/fortls"

I installed fortls on the original python environment instead of anaconda environment.

So one more question: if I installed the fortls on one anaconda env, need I open this anaconda env during all the time I use the
 **modern fortran**?

**Reply by gnikit** [reply, marked as answer]
Probably that would work. AFAIK if you activate anaconda (or any Python venv in a terminal) and open vscode from within in e.g.

```sh
python3 -m venv .venv
source ./.venv/bin/activate
pip install fortls --pre  # install if not installed already
code .
```

fortls (or the formatters) should be discoverable by the automatic mechanism.
